src/main/java/core/basesyntax/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.transaction.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    private ShopServiceImpl shopServiceImpl;
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int transactionQuantity = transaction.getQuantity();
+        int currentQuantity = shopServiceImpl.getQuantity(fruit);
+        if (transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {
+            if (transactionQuantity > currentQuantity) {
+                throw new RuntimeException("Not enough " + fruit + " in stock for purchase");
+            }
+            shopServiceImpl.addFruits(fruit, currentQuantity - transactionQuantity);
+        } else {
+            throw new RuntimeException("Unsupported operation for PurchaseHandler");
+        }

```

Comment:
The `if` statement checking if the operation is PURCHASE is not needed here. This handler should only be used for PURCHASE operations, so if it's used for another operation, it's a design error in your program. You should ensure that the correct handler is used for each operation in the place where you select the handler (usually with a map or a switch statement).